src/log.js

```diff
@@ -747,23 +747,55 @@ class MemLogger extends FormattedLoggerBase {
  *
  * @class
  * @param {Object} opts – Optional, named parameters
- * @param {string} [opts.level='silly'] The minimum log level to sent to loggly
+ * @param {string} [opts.level='silly'] The minimum log level to sent to the logger
+ * @param {Logger} [opts.logger = rootLogger] The helix logger to use
  * @param {Function} [opts.filter=identity] Will be given every log message to perform
  *   arbitrary transformations; must return either another valid message object or undefined
  *   (in which case the message will be dropped).
+ * @param {object} [opts.fields] Additional log fields to add to every log message.
  */
 class InterfaceBase {
+  /**
+   * Helper method for setting defaults recursively.
+   * todo: add to ferrum?
+   * @private
+   */
+  static _deepsetdefault(src, dst) {
```

Comment:
Please use shallow merging only; deep merging/setdefault *should* be something that the developer actively opts into.

Reply:
hmm, it would be cool if we could support both, maybe with a dotted notation?

```
{
   'data.id': 42
}
```

Reply:
Sure we could! Although then we would have to think about the edge cases (e.g. what if a field containing a dot is supposed to be set? We would have to start coming up with an escaping scheme…possible but complex).

Or we could just have people use the filter field…in which case this feature is already supported…

During the initial fields support PR I wrote I started off with a fields property; I did want it to have features like deep merging, deleting fields, applying functions to fields and so on…I even have a file implementing a grammar very similar to the dot notation you are suggesting lying around; I at some point abandoned the approach because I realized we already have a DSL for transforming objects…javascript

I agree it would be cooler! I just do not think this coolness is worth the added complexity for helix-log and the increased learning curve for our users…